### PR TITLE
Create learners document

### PR DESCRIPTION
**Pull request by aditidugar-zz**
Document to describe learning algorithms.

@GalOshri I will need help from your team to fill out the definitions/descriptions of the learners.

[Internal Review URL](https://review.docs.microsoft.com/en-us/dotnet/machine-learning/resources/learners?branch=pr-en-us-5698)

### CODE REVIEW

**Comment by aditidugar** [comment]
@CESARDELATORRE will be helping with these docs too. He is now a collaborator on my forked docs repo so he should be able to take it from here.

@GalOshri are you able to help review the table accuracy as a first step?

File: docs/machine-learning/resources/learners.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
+# Learning Algorithms
+The following table will help you understand the different learning algorithms (learners) included in ML.NET and the [machine learning tasks](tasks.md) that they can be used for. Click on the learner name for more in-depth explanations on each of these learning methods.
```
Comment (GalOshri) [inline docs/machine-learning/resources/learners.md]: Perhaps "the machine learning tasks where they can be used"?

File: docs/machine-learning/resources/learners.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+| Learner Name | Binary classification | Multi-class classification | Regression | Ranking | Clustering | Anomaly Detection |
+| --- |:---:|:---:|:---:|:---:|:---:|:---:|
+| [Averaged Perceptron](#averaged-perceptron) |:heavy_check_mark:||||||
+| [FastForest](#fastforest) |:heavy_check_mark:||:heavy_check_mark:||||
+| [FastTree](#fasttree)|:heavy_check_mark:||:heavy_check_mark:|:heavy_check_mark:|||
+| [Generalized Additive Models](#generalized-additive-models) |:heavy_check_mark:||:heavy_check_mark:||||
+| [K-Means++](#k-means) |||||:heavy_check_mark:||
+| [Linear SVM](#linear-svm) |:heavy_check_mark:||||||
+| [Logistic Regressor](#logistic-regressor) |:heavy_check_mark:|:heavy_check_mark:|||||
```
Comment (GalOshri) [inline docs/machine-learning/resources/learners.md]: The name of the Logistic Regression learner has been updated (should just be [Logistic Regression](https://github.com/dotnet/machinelearning/blob/1b08ae43887887abb478130e070dd8407c5e17b9/src/Microsoft.ML.StandardLearners/Standard/LogisticRegression/LogisticRegression.cs#L389))

File: docs/machine-learning/resources/learners.md [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+## Generalized Additive Models
+
+
+## K-Means++
+
+
+## Linear SVM
+
+
+## Logistic Regressor
```
Comment (GalOshri) [inline docs/machine-learning/resources/learners.md]: Each learner has description in the [docs](https://docs.microsoft.com/en-us/dotnet/api/microsoft.ml.trainers.logisticregressionbinaryclassifier?view=ml-dotnet) based on the description in the [code](https://github.com/dotnet/machinelearning/blob/1b08ae43887887abb478130e070dd8407c5e17b9/src/Microsoft.ML.StandardLearners/Standard/LogisticRegression/LogisticRegression.cs#L389). I'd recommend avoiding duplication if possible, linking to the other description, and improving the description in the code.

**Comment by GalOshri** [comment]
Looks good so far, thank you for creating this!

File: docs/machine-learning/resources/learners.md [hunk 4]
```diff
@@ -0,0 +1,67 @@
+# Learning Algorithms
+The following table will help you understand the different learning algorithms (learners) included in ML.NET and the [machine learning tasks](tasks.md) that they can be used for. Click on the learner name for more in-depth explanations on each of these learning methods.
+
+> [!NOTE]
+> ML.NET is currently in Preview. Not all learning algorithms are currently supported. To submit a request for a certain learner, please open an issue in the [dotnet/machinelearning repository](https://github.com/dotnet/machinelearning/issues).
+
+| Learner Name | Binary classification | Multi-class classification | Regression | Ranking | Clustering | Anomaly Detection |
```
Comment (OliaG) [inline docs/machine-learning/resources/learners.md]: In other docs we are spelling "Multiclass", not  "Multi-class". I'd suggest to be consistent. in ML world "Multiclass" seems to be more common (https://en.wikipedia.org/wiki/Multiclass_classification, http://www.mit.edu/~9.520/spring09/Classes/multiclass.pdf, https://www.coursera.org/learn/machine-learning/lecture/gFpiW/multiclass-classification) .

File: docs/machine-learning/resources/learners.md [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+| --- |:---:|:---:|:---:|:---:|:---:|:---:|
+| [Averaged Perceptron](#averaged-perceptron) |:heavy_check_mark:||||||
+| [FastForest](#fastforest) |:heavy_check_mark:||:heavy_check_mark:||||
+| [FastTree](#fasttree)|:heavy_check_mark:||:heavy_check_mark:|:heavy_check_mark:|||
+| [Generalized Additive Models](#generalized-additive-models) |:heavy_check_mark:||:heavy_check_mark:||||
+| [K-Means++](#k-means) |||||:heavy_check_mark:||
+| [Linear SVM](#linear-svm) |:heavy_check_mark:||||||
+| [Logistic Regressor](#logistic-regressor) |:heavy_check_mark:|:heavy_check_mark:|||||
+| [Naïve Bayes](#naïve-bayes) ||:heavy_check_mark:|||||
```
Comment (OliaG) [inline docs/machine-learning/resources/learners.md]: Too many dots above the "i"? :) In our code and in majority of textbooks it's "Naive", which is also easier to search for users.

**Pull request closed without merging**